common/src/Microsoft.Azure.IIoT.Agent.Framework/src/Agent/Default/Worker.cs

```diff
@@ -149,22 +157,24 @@ public class Worker : IWorker, IDisposable {
         /// <param name="sender"></param>
         private async void HeartbeatTimer_ElapsedAsync(object sender) {
 
-            await SendHeartbeatWithoutResetTimer();
-            Try.Op(() => _heartbeatTimer.Change(_heartbeatInterval, Timeout.InfiniteTimeSpan));
+            await SendHeartbeatWithoutResetTimer().ConfigureAwait(false);
+            if (!_cts.IsCancellationRequested) {
```

Comment:
`SendHeartbeatWithoutResetTimer()` also uses `_cts`. So maybe the call to it can also be under `if` ?

Reply:
right.